source/components/Bubble.js

```diff
@@ -75,6 +76,31 @@ class BubbleBase extends Component<Props, State> {
     }, 0);
   }
 
+  componentWillReceiveProps(nextProps: Props) {
```

Comment:
@MarcusHurney from what i can see the composition check is duplicated to all components … please refactor that to a helper function!

Reply:
@DominikGuzei

I should have been more explicit in my description of why this is necessary. Sorry for that, here it is: 

I purposefully needed to add this composition check to all components as well as `ThemeProvider` because let’s say you change the `themeOverrides` prop being passed to `ThemeProvider` dynamically as I do in its new storybook story that contains ProgressBar and a theme switcher button. The button gets pressed changing the value of `themeOverrides`. `ThemeProvider` will then recompose it’s two props `theme` and `themeOverrides` and the result is updated in `this.state.theme` which is used to construct the context object all components have access to. That context object with its newly composed `context.theme` property is passed to all components, but unless the components themselves do the composition check in `componentWillReceiveProps` to see if `props.theme`, `props.themeOverrides`, or `props.context` have changed, then their local state will not get updated and subsequently their themes will not dynamically update. So the check must be run on both `ThemeProvider` and the component level.

Another use case where this could be especially relevant is when each instance of a particular component, let’s say the Input component, has its own unique `props.themeOverrides` object. When `ThemeProvider` composes a new theme in context, the component level composition hasn’t taken place that will ultimately create a unique and final theme object in the local state of each Input because of the differences in each of their `themeOverrides` properties. 

If you have any suggestions on how to improve this composition processes, I’d be happy to refine this! The `themeOverrides` story in ThemeProvider’s storybook shows all of this in action and is where I originally found this duplicated check to be necessary.

Reply:
@MarcusHurney that's all fine but you can move the body of the function into a separate helper function which is called from `componentWillReceiveProps` and save hundreds of duplicated lines?

Reply:
Don't worry @MarcusHurney 😉 thanks for making the changes - looks much better now 🎉